cms/src/main/java/uk/nhs/digital/ps/modules/searchableFlagModule/DatasetProcessor.java

```diff
@@ -0,0 +1,74 @@
+package uk.nhs.digital.ps.modules.searchableFlagModule;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+
+public class DatasetProcessor extends AbstractProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(DatasetProcessor.class);
+
+    public void processNode(Node node, List<String> validStatuses) {
+        // find parent publication
+        Node parentPublication = findParentPublication(node);
+
+        if (parentPublication == null) {
+            // no parent publication found
+            log.error("Parent publication not found");
+            log.error("failed for node" + getRepoValue(node::getPath));
+            return;
+        }
+
+        // get common:searchable value
+        boolean searchable = getRepoValue(() -> parentPublication.getNode(parentPublication.getName()).getProperty("publicationsystem:PubliclyAccessible").getBoolean());
+
+        // apply it to dataset document(s)
+        streamChildNodes(node)
+            .filter((document) -> hasValidStateAndType(document, validStatuses,"publicationsystem:dataset"))
+            .forEach((document) -> setSearchableFlag(document, searchable));
+    }
+
+    private Node findParentPublication(Node node) {
+        Node publication = null;
+
+        try {
+            Node folder = node.getParent();
+            publication = findPublicationInFolder(folder);
+        } catch (RepositoryException ex) {
+            ex.printStackTrace();
```

Comment:
should log this with logger rather than printStackTrace()
(while you're there could rename to re from ex but that's just my preference 😄 )